spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -92,18 +95,26 @@
 	@Value("${spring.cloud.task.closecontext.enable:true}")
 	private Boolean closeContext;
 
+	@Value("${spring.cloud.task.executionid:}")
+	private Integer taskExecutionId;
+
+	@Value("${spring.cloud.task.external.executionid:#{null}}")
```

Comment:
Why does it fail in the shell?  That is the standard syntax for boot properties?  See here: http://docs.spring.io/spring-boot/docs/current/reference/html/common-application-properties.html